projects/aca-shared/rules/src/app.rules.spec.ts

```diff
@@ -1207,58 +1032,35 @@ describe('app.evaluators', () => {
     });
   });
 
-  describe('isTrashcanItemSelected', () => {
-    it('should return false when user is not in trashcan', () => {
-      context.navigation.url = '/personal-files/test';
-      expect(app.isTrashcanItemSelected(context)).toBeFalse();
-    });
-
-    it('should return false when nothing is selected', () => {
-      context.navigation.url = '/trashcan/test';
-      context.selection.isEmpty = true;
-      expect(app.isTrashcanItemSelected(context)).toBeFalse();
-    });
-
-    it('should return true when user is in trashcan and node is selected', () => {
-      context.navigation.url = '/trashcan/test';
-      context.selection.isEmpty = false;
-      expect(app.isTrashcanItemSelected(context)).toBeTrue();
-    });
-  });
-
-  describe('canViewFile', () => {
-    it('should return false when user is in trashcan', () => {
-      context.navigation.url = '/trashcan/test';
-      expect(app.canViewFile(context)).toBeFalse();
-    });
-
-    it('should return false when nothing is selected', () => {
-      context.navigation.url = '/personal-files/test';
-      context.selection.file = null;
-      expect(app.canViewFile(context)).toBeFalse();
+  describe('canToggleFileLock', () => {
+    beforeEach(() => {
+      context.profile = {} as any;
     });
 
-    it('should return true when user is not in trashcan and file is selected', () => {
-      context.navigation.url = '/personal-files/test';
-      context.selection.file = {} as any;
-      expect(app.canViewFile(context)).toBeTrue();
+    it('should return false when permission requirements are not met, regardless of file lock', () => {
+      context.selection.file = { entry: { properties: { 'cm:lockType': 'WRITE_LOCK', 'cm:lockOwner': { id: 'test' } } } } as any;
+      context.permissions = { check: () => false };
+      expect(app.canToggleFileLock(context)).toBeFalse();
     });
-  });
 
-  describe('canLeaveLibrary', () => {
-    it('should return false when no library is selected', () => {
-      context.selection.library = null;
-      expect(app.canLeaveLibrary(context)).toBeFalse();
+    it('should return true when file has no lock and permission requirements are met', () => {
+      context.selection.file = { entry: { properties: {} } } as any;
```

Comment:
same here, please use NodeEntry instead of any

Reply:
Replaced